integration-tests/scripts/copy-branch-to-repo-git.sh

```diff
@@ -0,0 +1,207 @@
+if [ -n "${DEBUG}" ]; then
+  echo "🐛 Destination repo API response: ${DEST_REPO_RESPONSE}"
+fi
+DEST_REPO_CHECK=$(echo "${DEST_REPO_RESPONSE}" | jq -r '.full_name // ""')
+if [ -z "${DEST_REPO_CHECK}" ]; then
+  echo "⚠️  Destination repository ${dest_repo} not found"
+  echo "   Attempting to create it automatically..."
+  
+  # Pass the full repository path to the create script (it now handles org/repo format)
+  if [ -n "${DEBUG}" ]; then
+    echo "🐛 Full dest_repo: ${dest_repo}"
+  fi
+  
+  # Get the directory where this script is located to find the helper script
+  SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+  CREATE_REPO_SCRIPT="${SCRIPT_DIR}/create-github-repo.sh"
+  
+  # Check if the helper script exists
+  if [ ! -f "${CREATE_REPO_SCRIPT}" ]; then
+    echo "🔴 error: repository creation helper script not found at ${CREATE_REPO_SCRIPT}"
+    echo "   Please ensure create-github-repo.sh is in the same directory as this script"
+    exit 1
+  fi
+  
+  # Create the repository using the helper script (pass full repo path)
+  echo "Creating repository ${dest_repo}..."
+  if [ -n "${DEBUG}" ]; then
+    echo "🐛 Running: ${CREATE_REPO_SCRIPT} ${dest_repo} 'Automatically created for branch copy from ${source_repo}' false"
+    DEBUG=1 "${CREATE_REPO_SCRIPT}" "${dest_repo}" "Automatically created for branch copy from ${source_repo}" false
+  else
+    "${CREATE_REPO_SCRIPT}" "${dest_repo}" "Automatically created for branch copy from ${source_repo}" false
+  fi
+  
+  if [ $? -ne 0 ]; then
+    echo "🔴 error: failed to create destination repository ${dest_repo}"
+    exit 1
+  fi
+  
+  # Verify the repository was created successfully
+  echo "Re-verifying destination repository ${dest_repo}..."
+  DEST_REPO_RECHECK_RESPONSE=$(curl -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/${dest_repo} 2> /dev/null)
+  if [ -n "${DEBUG}" ]; then
+    echo "🐛 Destination repo recheck API response: ${DEST_REPO_RECHECK_RESPONSE}"
+  fi
+  DEST_REPO_CHECK=$(echo "${DEST_REPO_RECHECK_RESPONSE}" | jq -r '.full_name // ""')
+  if [ -z "${DEST_REPO_CHECK}" ]; then
+    echo "🔴 error: destination repository ${dest_repo} still not accessible after creation"
+    if [ -n "${DEBUG}" ]; then
+      echo "🐛 Recheck response: ${DEST_REPO_RECHECK_RESPONSE}"
+    fi
+    exit 1
+  fi
+  
+  echo "✅ Repository ${dest_repo} created successfully"
+fi
+
+echo "✅ Both repositories are accessible"
+
+# Create a temporary directory for the operation
+TEMP_DIR=$(mktemp -d)
+echo "Using temporary directory: $TEMP_DIR"
+
+# Cleanup function
+cleanup() {
+  echo "Cleaning up temporary directory..."
+  rm -rf "$TEMP_DIR"
+}
+trap cleanup EXIT
+
+cd "$TEMP_DIR"
+
+# Clone the source repository
+echo "Cloning source repository ${source_repo}..."
+git clone "https://${GITHUB_TOKEN}@github.com/${source_repo}.git" source-repo
+cd source-repo
+
+# Check if source branch exists
```

Comment:
I don't think this is needed either. When you go to `git checkout` it tells you if it isn't there
```
release-service]$ git checkout abc
error: pathspec 'abc' did not match any file(s) known to git
```

Reply:
fixed.